fixtures/node_annotate.py

```diff
@@ -0,0 +1,115 @@
+import py
+import pytest
+import csv
+import yaml
+import os
+from operator import itemgetter
+
+ROOT = py.path.local(__file__).dirpath().dirpath()
+
+
+class MarkFromMap(object):
+    def __init__(self, mark_map):
+        self.mark_map = mark_map
+
+    def pytest_itemcollected(self, item):
+        mark = self.mark_map.get(item.nodeid)
+        if mark is not None:
+            item.add_marker(mark)
+
+    @classmethod
+    def from_parsed_list(cls, parsed, key, map_value):
+        data = dict(map(itemgetter('id', key), parsed))
+        mark_map = dict((k, map_value(v)) for k, v in data.items())
+        return cls(mark_map)
+
+
+def pytest_configure(config):
+    path = config.getini('cfme_annotations_path')
+    if path:
+        to_parse = ROOT.join(path)
+        parsed = parse(to_parse)
+    else:
+        parsed = []
+    config.pluginmanager.register(
+        MarkFromMap.from_parsed_list(parsed, 'tier', pytest.mark.tier))
+    config.pluginmanager.register(
+        MarkFromMap.from_parsed_list(parsed, 'type', pytest.mark.__getattr__))
+
+
+def pytest_addoption(parser):
+    group = parser.getgroup('cfme')
+    group.addoption(
+        '--tier', type=int, action='append',
+        help='only run tests of the given tiers')
+    parser.addini(
+        'cfme_annotations_path',
+        help="path to a cfme annotations file")
+
+
+def tier_matches(item, tiers):
+    mark = item.keywords.get('tier')
+    if mark is None or getattr(mark, 'args', None) is None:
+        return True
+    return mark.args[0] in tiers
+
+
+def pytest_collection_modifyitems(config, items):
+    tiers = config.getoption('tier')
+    if not tiers:
+        return
+    # TODO(rpfannsc) trim after pytest #1373 is done
+    keep, discard = [], []
+    for item in items:
+        if tier_matches(item, tiers):
+            keep.append(item)
+        else:
+            discard.append(item)
+
+    items[:] = keep
+    # TODO(rpfannsc) add a reason after pytest #1372 is fixed
+    config.hook.pytest_deselected(items=discard)
+
+
+def generate_nodeid(mapping):
```

Comment:
the tiers are added via the MarkFromMap plugin class, i add 2 of those to the pluginmanager at the end of the configure hook